### PR TITLE
#162949236 Get user's reading stats

### PR DESCRIPTION
**Pull request by nesh-dev**
#### What does this PR do?
Users list all the articles they have viewed
Handle tests for the functionality

#### Description of Task to be completed?

* Have the following working

- [x] List of all the users read articles

Use:
 * GET`api/stats/

Payload:

#### How should this be manually tested?

1. Clone the repo: `git clone https://github.com/andela/ah-the-unsullied.git`
2. Cd into the cloned repo 
3. Git pull branch ft-read-stats-162949236
4. Set up a virtual environment and install requirements:
      ```
            python3 -m venv venv
            source venv/bin/activate
            pip3 install -r requirements.txt

     ```
5. Set up a .env file in the same level as the venv file and add the following:
      ```
         export SECRET_KEY=""
       export DATABASE_URL=""
      export DJANGO_SETTINGS_MODULE=''
export CLOUDINARY_NAME=''
export CLOUDINARY_KEY=''
export CLOUDINARY_SECRET=''
export HOST_PASSWORD=""
export HOST_USER=""
export SMTP_HOST=''
export EMAIL_PORT='587'
export FACEBOOK_KEY=''
export FACEBOOK_SECRET=''
export TWITTER_KEY=''
export TWITTER_SECRET_KEY=''
export GOOGLE_KEY=''
export GOOGLE_SECRET=''
export ACCESS_TOKEN=''
export SECRET_TOKEN=''
      ```
6. Then run: 
     ``` 
            source .env 
            ./manage.py migrate
            ./manage.py runserver
      ```
7. To test you need to register and login via postman.

* Register endpoint `/api/users`
* Login endpoint `/api/users/login`

#### What are the relevant pivotal tracker stories?
[#162949236](https://www.pivotaltracker.com/n/projects/2235173/stories/162949236)
#### Screenshots (if appropriate)
screenshot 2019-02-04 at 19 49 00

#### Questions:

### CODE REVIEW

File: authors/apps/articles/views/articles.py [hunk 1]
```diff
@@ -11,7 +11,12 @@
 from django.contrib.contenttypes.models import ContentType
 
 # local imports
+
 from authors.apps.articles.models import Article, LikeDislike
+
+from authors.apps.articles.models import Article
```
Comment (Allan690) [inline authors/apps/articles/views/articles.py]: You have a similar import at line 15.

File: authors/apps/reading_stats/models.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    user = models.ForeignKey(AUTH_USER_MODEL, on_delete=models.CASCADE)
+    article = models.ForeignKey(Article, on_delete=models.CASCADE)
+    is_read = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateField(auto_now=True)
+
+    def __str__(self):
+        return self.article.title
+
```
Comment (Allan690) [inline authors/apps/reading_stats/models.py]: Kindly remove this extra line

File: authors/apps/reading_stats/serializers.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    def get_article(self, stats_object):
+        return {
+            "article": stats_object.article.title,
+            "slug": stats_object.article.slug
+        }
+
+    class Meta:
+        model = ReadingStats
+        fields = ['article', 'user']
+
```
Comment (Allan690) [inline authors/apps/reading_stats/serializers.py]: Remove this line also

File: authors/urls.py [hunk 4]
```diff
@@ -33,5 +33,9 @@
     path('auth/', include('rest_framework_social_oauth2.urls')),
     path('api/', include('authors.apps.ratings.urls', namespace='ratings')),
     path('api/tags', TagView.as_view()),
-    path('', schema_view)
+    path('', schema_view),
+
```
Comment (Allan690) [inline authors/urls.py]: Kindly remove the space between the lines

**Review by Allan690 — DISMISSED** [review]

**Review by sylviawanjiku — APPROVED** [review]

**Review by Allan690 — APPROVED** [review]

**Pull request merged**